### PR TITLE
Fixes #8911: missing rudder_expected_reports.csv.res when starting the agent for the first time after an update - technique changes - 3.1 branch

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8911

### CODE REVIEW

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -46,7 +46,7 @@ bundle common va
 {
   vars:
 
-      "end" slist => { "restart_services", "endExecution" };
+      "end" slist => { "restart_services", "_clean_current_expected_reports_file", "endExecution" };
```
Comment (peckpeck) [inline techniques/system/common/1.0/promises.st]: Where is this method ?

File: techniques/system/common/1.0/promises.st [hunk 2]
```diff
@@ -654,6 +654,11 @@ bundle agent garbage_collection
         comment     => "Rotate file if above specified size",
         rename      => rotate("10"),
         file_select => bigger_than("1M");
+
+  methods:
+      # Directly call the cleanup bundle from ncf's logger_rudder
+      "clean old expected reports files" usebundle => _clean_old_expected_reports_file("&CFENGINE_OUTPUTS_TTL&");
```
Comment (peckpeck) [inline techniques/system/common/1.0/promises.st]: And this one ?

File: techniques/system/common/1.0/promises.st (hunk 1 above)
Comment (jooooooon) [inline techniques/system/common/1.0/promises.st]: In ncf. See normation/ncf#423

**Comment by amousset** [comment]
This depends on Normation/ncf#423, don't merge yet

**Comment by amousset** [comment]
Done when merging #1013, closing.

**Pull request closed without merging**